Std/Data/HashMap/Basic.lean

```diff
@@ -291,51 +317,128 @@ replaced with `a => b`.
 
 /--
 Removes key `a` from the map. If it does not exist in the map, the map is returned unchanged.
+```
+def hashMap := ofList [("one", 1), ("two", 2)]
+hashMap.erase "one" = {"two" ↦ 2}
+hashMap.erase "three" = {"one" ↦ 1, "two" ↦ 2}
+```
 -/
 @[inline] def erase (self : HashMap α β) (a : α) : HashMap α β := ⟨self.1.erase a, self.2.erase⟩
 
 /--
 Performs an in-place edit of the value, ensuring that the value is used linearly.
 The function `f` is passed the original key of the entry, along with the value in the map.
+```
+(ofList [("one", 1), ("two", 2)]).modify "one" (fun _ v => v + 1) = {"one" ↦ 2, "two" ↦ 2}
+```
 -/
 def modify (self : HashMap α β) (a : α) (f : α → β → β) : HashMap α β :=
   ⟨self.1.modify a f, self.2.modify⟩
 
-/-- Given a key `a`, returns a key-value pair in the map whose key compares equal to `a`. -/
+/--
+Given a key `a`, returns a key-value pair in the map whose key compares equal to `a`.
+```
+def hashMap := ofList [("one", 1), ("two", 2)]
+hashMap.findEntry? "one" = some ("one", 1)
+hashMap.findEntry? "three" = none
+```
+-/
```

Comment:
There is a subtlety in this function that is only demonstrated if the key type has a BEq instance which is weaker than `=`: the returned key is not always the same as the input key, it is only `==` to it. Unfortunately I don't think there are any builtin types which have this behavior. Are there examples of this in the RBMap or List docs? I think the example might have been a pair where the equality is defined to only look at the first component.

Reply:
There is no example in [the doc for the `findEntry?` function for RBMap](https://leanprover-community.github.io/mathlib4_docs/Std/Data/RBMap/Basic.html#Std.RBMap.findEntry%3F) and I don't see a `findEntry?` for List. Would it be in another function?

There are some functions in `RBMap` like [this](https://leanprover-community.github.io/mathlib4_docs/Std/Data/RBMap/Basic.html#Std.RBMap.instBEqRBMap) one which mentions the compatibility of the ordering `cmp` and `=`.

Reply:
I can't find the example I was thinking of. In the absence of good demonstrative examples we can just have a line of explanatory text.